qcodes/dataset/data_set.py

```diff
@@ -1248,6 +1248,17 @@ def _load_to_xarray_dataset(self, data: ParameterData) -> "xr.Dataset":
 
         xrdataset.attrs["sample_name"] = self.sample_name
         xrdataset.attrs["exp_name"] = self.exp_name
+        xrdataset.attrs["guid"] = self.guid
+        xrdataset.attrs["run_timestamp"] = self.run_timestamp_raw
+        xrdataset.attrs["completed_timestamp"] = self.completed_timestamp_raw
+        xrdataset.attrs["parameters"] = self.parameters
+        xrdataset.attrs["run_description"] = self.description
+        xrdataset.attrs["snapshot"] = self.snapshot
+        xrdataset.attrs["captured_run_id"] = self.captured_run_id
+        xrdataset.attrs["captured_counter"] = self.captured_counter
+
+        for metadata_tag, metadata in self._metadata.keys():
```

Comment:
Shall we not mix metadata with other keys and store metadata dict under “metadata” key?

Reply:
Not really sure this layout matches the way its stored in the dataset but I could also see the argument for nesting it one more level. I don't think we should call it metadata however since all of these columns are really metadata. Its more like `user added custom metadata` but what is a good name for that?

Reply:
agree with you! (and "this layout matches the way its stored in the dataset" i think is not relevant really). what about `extra_metadata` ? :)

Reply:
So it looks like attrs do not allow None elements. We should normalize any none out of fixed columns (easy enough I think since we can use json str for description and snapshot but we also need to recursively normalize the metadata field

Reply:
or we remove the None-valued metadata items?

Reply:
well that is one form of normalization